### PR TITLE
feat: add prometheus.write.queue bearer token auth

### LINKED ISSUE #1913: Add bearer auth for prometheus.write.queue
### Request

Currently, prometheus.write.queue only supports Basic Auth, would be nice to also have bearer auth there, same way as in prometheus.remote_write component.

### Use case

For us, we are using Prometheus managed solution, which only allows using bearer auth, so basic auth won't work here, therefore we need to be able to use bearer auth.

### PR DESCRIPTION
**Pull request by freak12techno**
#### Which issue(s) this PR fixes

Fixes #1913 

#### PR Checklist

- [x] CHANGELOG.md updated
- [x] Documentation added
- [ ] Tests updated
- [x] Config converters updated

### CODE REVIEW

**Comment by freak12techno** [comment]
@mattdurham mind taking a look?

**Comment by mattdurham** [comment]
Would like to see a validation so that you cannot set both bearer token and basic auth, with a test to catch that. Otherwise looks good!

File: internal/component/prometheus/write/queue/types/network.go [hunk 1]
```diff
@@ -2,6 +2,7 @@ package types
 
 import (
 	"context"
+	"github.com/grafana/alloy/syntax/alloytypes"
```
Comment (mattdurham) [inline internal/component/prometheus/write/queue/types/network.go:5]: This likely fails the linter, generally "core" golang imports are separated from other imports.

Comment (mattdurham) [reply internal/component/prometheus/write/queue/types/network.go:5]: It didnt..coolio :)

File: docs/sources/reference/components/prometheus/prometheus.write.queue.md [hunk 2]
```diff
@@ -83,16 +83,17 @@ The `endpoint` block describes a single location to send metrics to. Multiple
...
-`batch_count` | `uint` | How many series to queue in each queue.                          | `1000` | no
-`flush_interval` | `duration` | How often to wait until sending if `batch_count` is not trigger. | `1s` | no
-`parallelism` | `uint` | How many parallel batches to write.                              | 10 | no
-`external_labels` | `map(string)` | Labels to add to metrics sent over the network.                  | | no
+Name | Type | Description                                                        | Default | Required
+---- | ---- |--------------------------------------------------------------------| ------ | --------
+`url` | `string` | Full URL to send metrics to.                                       | | yes
+`bearer_token` | `secret` | Bearer token to authenticate with.                                 |  | no
+`write_timeout` | `duration` | Timeout for requests made to the URL.                              | `"30s"` | no
+`retry_backoff` | `duration` | How often to wait between retries.                                 | `1s` | no
```
Comment (clayton-cornell) [inline docs/sources/reference/components/prometheus/prometheus.write.queue.md:91]: How often or how long?

File: docs/sources/reference/components/prometheus/prometheus.write.queue.md [hunk 3]
```diff
@@ -83,16 +83,17 @@ The `endpoint` block describes a single location to send metrics to. Multiple
...
+`max_retry_attempts` | Maximum number of retries before dropping the batch. | `0`                                                                | no
+`batch_count` | `uint` | How many series to queue in each queue.                            | `1000` | no
+`flush_interval` | `duration` | How often to wait until sending if `batch_count` is not triggered. | `1s` | no
```
Comment (clayton-cornell) [inline docs/sources/reference/components/prometheus/prometheus.write.queue.md:94]: Same question.. often or long? If we say "how often" it doesn't make sense to me.. Wait 6 times (would be how often) vs Wait 2 minutes (would be how long).

**Review by clayton-cornell — DISMISSED** [review]

File: docs/sources/reference/components/prometheus/prometheus.write.queue.md (hunk 2 above)
Comment (mattdurham) [reply docs/sources/reference/components/prometheus/prometheus.write.queue.md:91]: @clayton-cornell I dont think he changed most of these, I think adding bearer_token triggered them to look different. I can create a separate PR for these doc comments related to backoff and flush interval.

Comment (freak12techno) [reply docs/sources/reference/components/prometheus/prometheus.write.queue.md:91]: Yeah I didn't, it's my IDE that apparently tried to format it, only thing I changed is adding this line and also fixing the typo (trigger -> triggered) in another one.

Comment (freak12techno) [reply docs/sources/reference/components/prometheus/prometheus.write.queue.md:91]: I might change it here as well though if you prefer

**Comment by mattdurham** [comment]
@freak12techno I will follow up on the docs stuff later, has long as @clayton-cornell is good with the bearer token text.

File: docs/sources/reference/components/prometheus/prometheus.write.queue.md (hunk 2 above)
Comment (mattdurham) [reply docs/sources/reference/components/prometheus/prometheus.write.queue.md:91]: Handling it in #1917

**Review by clayton-cornell — APPROVED** [review]
LGTM, no blocks from docs side

**Review by mattdurham — APPROVED** [review]
Lgtm

**Comment by freak12techno** [comment]
okay @mattdurham I didn't get a chance to add tests + validations before this got merged :( should I make another PR adding it, or would you add it in your PR?

**Comment by mattdurham** [comment]
I can handle it, I jumped the gun on it

**Pull request merged**